src/converters/app-converter.ts

```diff
@@ -169,6 +164,72 @@ export class StackConverter extends ArtifactConverter {
+     * "registers" it to the StackConverter. The StackConverter keeps track of all of the
+     * resources that were mapped from the CFN resources in the stack. It uses this information
+     * to resolve references to other resources in the stack.
+     *
+     * Typically there will be a couple of scenarios for how a CFN resource maps to Pulumi resources:
+     *
+     * 1. The CFN resource maps to a single Pulumi aws-native resource
+     *    This is the most straightforward case because everything maps directly, including the attributes
+     *
+     * 2. The CFN resource maps to single Pulumi aws resources
+     *    In this case the mapping is to a single resource, but some times the attributes available
+     *    on the CFN resource do not map to the attributes on the Pulumi resource. In these cases
+     *    the mapping can return custom attributes that are later available when references are resolved
+     *
+     * 3. The CFN resource maps to multiple Pulumi aws resources
+     *    One example would be the AWS::IAM::Policy resource which in CFN includes Role, Group, and User
+     *    Polices, but in Pulumi aws these are broken out into separate resources. In that case the "main"
+     *    resource would be the Policy and the other resources would be added to the resources array.
+     *    The critical point here is that the "main" resource needs to have a logicalId that matches the
+     *    logicalId of the CFN resource, while the other supporting resources must have different logicalIds.
+     *
+     * @param mapped - The Pulumi Resource that was mapped from the CFN resource
+     * @param node - The GraphNode that represents the CFN Resource
+     */
+    private registerResource(mapped: ResourceMapping, node: GraphNode): void {
+        const cfn = node.resource;
+        // This should always be set because we only call this function when it is, but
+        // TypeScript doesn't know that.
+        if (!cfn) {
+            throw new Error('Cannot map a resource without a CloudFormation resource');
+        }
+        const mainResource: ResourceAttributeMapping | undefined = Array.isArray(mapped)
+            ? mapped.find((res) => res.logicalId === node.logicalId)
+            : pulumi.Resource.isInstance(mapped)
+            ? { resource: mapped }
+            : mapped;
+        if (!mainResource) {
+            throw new Error(
+                `Resource mapping for ${node.logicalId} of type ${cfn.Type} did not return a primary resource. \n` +
+                    'Examine your code in "remapCloudControlResource"',
+            );
+        }
+        const resourceMapping: Mapping<pulumi.Resource> = {
+            resource: mainResource.resource,
+            attributes: mainResource.attributes,
+            resourceType: cfn.Type,
+            otherResources: [],
+        };
+        this.constructs.set(node.construct, mainResource.resource);
+        if (Array.isArray(mapped)) {
```

Comment:
What do you think about moving this above the declaration of `resourceMapping`? Then instead of having to mutate the `resourceMapping.otherResources` array in the `forEach`, we could do a `map` instead and just set the `otherResources`